README.md

```diff
@@ -1,2 +1,59 @@
 # action-cli-login
-GitHub action to login to a tenant using Office 365 CLI 
+GitHub action to login to a tenant using Office 365 CLI.
+
+![Office 365 CLI Login](./images/cli-login.png)
+
+This GitHub Action (created using Docker container) uses [Office 365 CLI](https://pnp.github.io/office365-cli/), specifically the [login command](https://pnp.github.io/office365-cli/cmd/login), to allow you log in to Office 365.
+
+## Usage
+### Pre-requisites
+Create a workflow `.yml` file in your `.github/workflows` directory. An [example workflow](#example-workflow---office-365-cli-login) is available below. For more information, reference the GitHub Help Documentation for [Creating a workflow file](https://help.github.com/en/articles/configuring-a-workflow#creating-a-workflow-file).
+
+### Inputs
+- `ADMIN_USERNAME` : **Required** Username (email address of the admin)
+- `ADMIN_PASSWORD` : **Required** Password of the admin
+
+#### Optional requirement
+Since this action requires user name and password which are sensitive pieces of information, it would be ideal to store them securely. We can achieve this in a GitHub repo by using [secrets](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets). So, click on `settings` tab in your repo and add 2 new secrets:
+- `adminUsername` - store the admin user name in this (e.g. user@contoso.onmicrosoft.com)
+- `adminPassword` - store the password of that user in this.
+These secrets are encrypted and can only be used by GitHub actions. 
+
+### Example workflow - Office 365 CLI Login
+On every `push` build the code and then login to Office 365 before deploying.
+
+```yaml
+name: SPFx CICD with O365 CLI
+
+on: [push]
+
+jobs:
+  build:
+    ##
+    ## Build code omitted
+    ##
+        
+  deploy:
+    needs: build
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [10.x]
+    
+    steps:
+    
+    ##
+    ## Code to get the package omitted
+    ##
+
+    # Office 365 cli login action
+    - name: Login to tenant
+      uses: pnp/action-cli-login@v1
```

Comment:
@anoopt , is the expectation to run another action after the login action is completed. 

I am not quite sure how it works that is why I am asking questions. My assumptions is that 
- you will perform Office 365 CLI login using the current docker container
- once you run the next possible action, will the credential persist in a session? (since the other possible next action would run on different docker container, and the first will possibly be disposed)

I am asking because I really do not know how it works, and I am seeking for quick answer. Will dig deeper into this in the next days, but if you have quick answer, I can benefit form it.

Reply:
@VelinGeorgiev - No worries. Even I had the same doubt earlier. Reading the details mentioned [here](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/virtual-environments-for-github-hosted-runners) helped. So all the steps in a job run on the same virtual machine thereby all the steps share information. This means in our case, we login and then perform the O365 CLI actions on the same virtual machine. Hope this info is helpful. 

Also, please find below a sample workflow with all the O365 CLI actions (this is part of documentation in O365 CLI repo)

```yaml
name: SPFx CICD with Office 365 CLI
on: push
# Variables that are available for all the jobs in the workflow
env:
  SPPKG_FILE_NAME: ${{ 'package-name.sppkg' }}
  SEND_MAIL: false
jobs:
  build-and-deploy:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        node-version: [10.x]
    
    steps:
    # Checkout code
    - name: Checkout code
      uses: actions/checkout@v1
      
    # Setup node.js runtime
    - name: Use Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v1
      with:
        node-version: ${{ matrix.node-version }}
    
    # npm install
    - name: Run npm install
      run: npm install
    
    # gulp bundle and package solution
    - name: Bundle and package
      run: |
        gulp bundle --ship
        gulp package-solution --ship    
    
    # Login to tenant using action-cli-login
    - name: Office 365 CLI Login
      uses: pnp/action-cli-login@v1
      env:
        ADMIN_USERNAME:  ${{ secrets.adminUsername }}
        ADMIN_PASSWORD:  ${{ secrets.adminPassword }}
    
    # Deploy package to tenant using action-cli-deploy
    - name: Office 365 CLI Deploy
      uses: pnp/action-cli-deploy@v1
      env:
        APP_FILE_PATH: sharepoint/solution/${{ env.SPPKG_FILE_NAME }}
    
    # Send an email using action-cli-runscript
    - name: Office 365 CLI Send email
      uses: pnp/action-cli-runscript@v1
      env:
         O365_CLI_SCRIPT: o365 spo mail send --webUrl https://contoso.sharepoint.com/sites/teamsite --to 'user@contoso.onmicrosoft.com' --subject 'Deployment done' --body '<h2>Office 365 CLI</h2> <p>The deployment is complete.</p> <br/> Email sent via Office 365 CLI GitHub Action.'
      if: env.SEND_MAIL == true
```